### PR TITLE
GH-22 Add system environment support to token and logzioUrl parameters

### PR DESCRIPTION
**Pull request by asafalima**

### CODE REVIEW

**Comment by roiravhon** [comment]
LGTM.. @asafm ?

**Comment by roiravhon** [comment]
@asafalima Please also increment the verison in the pom.xml, the README and add release notes

File: src/main/java/io/logz/logback/LogzioSender.java [hunk 1]
```diff
@@ -67,11 +67,11 @@ private LogzioSender(String logzioToken, String logzioType, int drainTimeout, in
                         boolean addHostname, String additionalFields, int gcPersistedQueueFilesIntervalSeconds)
             throws IllegalArgumentException {
 
-        this.logzioToken = logzioToken;
+        this.logzioToken = getValueFromSystemEnvironmentIfNeeded(logzioToken);
         this.logzioType = logzioType;
         this.drainTimeout = drainTimeout;
         this.fsPercentThreshold = fsPercentThreshold;
-        this.logzioUrl = logzioUrl;
+        this.logzioUrl = getValueFromSystemEnvironmentIfNeeded(logzioUrl);
```
Comment (asafm) [inline src/main/java/io/logz/logback/LogzioSender.java]: 👍 
We have this logic also in additionalFields. Can you please re-use this function there as well?

File: src/test/java/io/logz/LogzioSenderTest.java [hunk 2]
```diff
@@ -417,4 +417,24 @@ public void testContextReset() throws Exception {
         logRequest = assertLogReceivedByMessage(message2);
         assertLogReceivedIs(logRequest, token, type, loggerName, Level.WARN);
     }
+
+    @Test
+    public void testTokenAndLogzioUrlFromSystemEnvironment() {
```
Comment (asafm) [inline src/test/java/io/logz/LogzioSenderTest.java]: 👍

File: src/main/java/io/logz/logback/LogzioSender.java [hunk 3]
```diff
@@ -67,11 +67,11 @@ private LogzioSender(String logzioToken, String logzioType, int drainTimeout, in
                         boolean addHostname, String additionalFields, int gcPersistedQueueFilesIntervalSeconds)
             throws IllegalArgumentException {
 
-        this.logzioToken = logzioToken;
+        this.logzioToken = getValueFromSystemEnvironmentIfNeeded(logzioToken);
```
Comment (asafm) [inline src/main/java/io/logz/logback/LogzioSender.java]: Can you please document this field support env variables and how in readme?
For both fields you've modified.

File: README.md [hunk 4]
```diff
@@ -35,7 +35,7 @@ This appender uses [BigQueue](https://github.com/bulldog2011/bigqueue) implement
 ### Parameters
 | Parameter          | Default                              | Explained  |
 | ------------------ | ------------------------------------ | ----- |
-| **token**              | *None*                                 | Your Logz.io token, which can be found under "settings" in your account |
+| **token**              | *None*                                 | Your Logz.io token, which can be found under "settings" in your account, If the value begins with `$` then the the appender look for environment variable with the name specified. For example: `$LOGZIO_TOKEN`will look for env variable named `LOGZIO_TOKEN` |
```
Comment (roiravhon) [inline README.md]: If the value begins with `$` then the appender looks for an environment variable with the name specified. For example: `$LOGZIO_TOKEN` will look for environment variable named `LOGZIO_TOKEN`

And can you please also add a documentation for logzioUrl ?

**Pull request merged**